src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,49 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final int EMPLOYEE_DATE_INDEX = 0;
+    private static final int EMPLOYEE_NAME_INDEX = 1;
+    private static final int EMPLOYEE_HOUR_INDEX = 2;
+    private static final int EMPLOYEE_PRICE_INDEX = 3;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        String[] employeeData;
+        int salary;
+        StringBuilder builder = new StringBuilder("Report for period "
+                + dateFrom + " - " + dateTo);
+
+        for (String name : names) {
+            salary = 0;
+            for (String record : data) {
+                employeeData = record.split(" ");
+
+                if (employeeData[EMPLOYEE_NAME_INDEX].equals(name)
+                        && isDateBetween(employeeData[EMPLOYEE_DATE_INDEX], dateFrom, dateTo)) {
```

Comment:
dateFrom, dateTo - should be parsed **before** loops

Reply:
Could you explain in more detail what that means "Should be parsed before loops", because I don't understand how to do it otherwise

Reply:
Converting `String dateFrom` to LocalDate object using `LocalDate.parse` method is **parsing operation**.
How many times are we going to execute this operation? One time is good enough I believe. But you do it inside the loop...